backend/controller/dal/notify.go

```diff
@@ -97,7 +97,7 @@ func (d *DAL) PollDeployments(ctx context.Context) {
 				d.DeploymentChanges.Publish(DeploymentNotification{
 					Message: optional.Some(deployment),
 				})
-			} else if !bytes.Equal(previousState.schemaHash, state.schemaHash) || previousState.minReplicas != state.minReplicas {
+			} else if !bytes.Equal(previousState.schemaHash, state.schemaHash) || previousState.minReplicas != state.minReplicas || !bytes.Equal(previousState.Key.Suffix, state.Key.Suffix) {
```

Comment:
I think at some point the idea was to compare the artefact hashes too, but that doesn't seem to have materialised.

Reply:
Do they form part of the suffix?

Reply:
Oh! Maybe they do, I can't 100% recall but I think they might yeah!